src/quic/tlscontext.h

```diff
@@ -0,0 +1,176 @@
+#pragma once
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+#if HAVE_OPENSSL && NODE_OPENSSL_HAS_QUIC
+
+#include <base_object.h>
+#include <crypto/crypto_context.h>
+#include <crypto/crypto_keys.h>
+#include <memory_tracker.h>
+#include <ngtcp2/ngtcp2_crypto.h>
+#include "bindingdata.h"
+#include "data.h"
+#include "sessionticket.h"
+
+namespace node {
+namespace quic {
+
+class Session;
+
+// Every QUIC Session has exactly one TLSContext that maintains the state
+// of the TLS handshake and negotiated cipher keys after the handshake has
+// been completed. It is separated out from the main Session class only as a
+// convenience to help make the code more maintainable and understandable.
+class TLSContext final : public MemoryRetainer {
+ public:
+  static constexpr auto DEFAULT_CIPHERS = "TLS_AES_128_GCM_SHA256:"
+                                          "TLS_AES_256_GCM_SHA384:"
+                                          "TLS_CHACHA20_POLY1305_"
+                                          "SHA256:TLS_AES_128_CCM_SHA256";
+  static constexpr auto DEFAULT_GROUPS = "X25519:P-256:P-384:P-521";
```

Comment:
Does it make sense for these to be separate from the TLS implementation's default ciphers and curves/groups?

Reply:
I think for now, yes. Not all of the TLS implementation's default ciphers and curves are appropriate for quic. I'd rather keep these lists separate for now and later we can explore other options.